unlock-app/src/hooks/metadata.ts

```diff
@@ -18,6 +18,13 @@ export const useUpdateMetadata = ({
   return useMutation(
     ['updateMetadata', network, lockAddress, keyId],
     async (metadata: Metadata): Promise<Partial<Metadata>> => {
+      const slug = metadata.slug
+      // save slug for page url
+      if (slug && network && lockAddress) {
+        await storage.saveLockSetting(network, lockAddress, {
+          slug,
+        })
+      }
```

Comment:
Seems an odd place to put it here. Shouldn't it be in a separate lock settings hook.